packages/ui-react/src/components/Account/Account.tsx

```diff
@@ -199,64 +176,54 @@ const Account = ({ panelClassName, panelHeaderClassName, canUpdateEmail = true }
-                firstName: values.firstName || '',
-                lastName: values.lastName || '',
-                metadata: {
-                  ...values.metadata,
-                  ...formatConsentsToRegisterFields(consents),
-                  consents: JSON.stringify(consents),
-                },
-              });
+            onSubmit: async (values) => {
+              if (registrationFields) {
+                await accountController.updateRegistrationFieldsValues(registrationFields.fields, values.registrationFieldsValues);
+                return;
+              }
+              throw new Error('Registration fields are not loaded');
             },
             content: (section) => (
               <>
                 <h1 className={styles.hideUntilFocus}>{t('nav.account')}</h1>
-                <TextField
-                  name="firstName"
-                  label={t('account.firstname')}
-                  value={section.values.firstName || ''}
-                  onChange={section.onChange}
-                  error={!!section.errors?.firstName}
-                  helperText={section.errors?.firstName}
-                  disabled={section.isBusy}
-                  editing={section.isEditing}
-                />
-                <TextField
-                  name="lastName"
-                  label={t('account.lastname')}
-                  value={section.values.lastName || ''}
-                  onChange={section.onChange}
-                  error={!!section.errors?.lastName}
-                  helperText={section.errors?.lastName}
-                  disabled={section.isBusy}
-                  editing={section.isEditing}
-                />
+                {registrationFields?.fields.map((field) => (
+                  <CustomRegisterField
+                    key={field.name}
+                    type={field.type}
+                    name={`registrationFieldsValues.${field.name}`}
+                    value={section.values.registrationFieldsValues[field.name] || ''}
+                    onChange={section.onChange}
+                    placeholder={field.placeholder}
+                    disabled={!section.isEditing || section.isBusy}
+                    label={field.label}
+                    required={field.required}
+                    options={field.options}
+                  />
+                ))}
               </>
             ),
           }),
           formSection({
             label: t('account.email'),
-            onSubmit: (values) =>
-              accountController.updateUser({
-                email: values.email || '',
+            onSubmit: async (values) => {
+              await accountController.updateUser({
+                email: values.customer?.email || '',
                 confirmationPassword: values.confirmationPassword,
-              }),
-            canSave: (values) => !!(values.email && values.confirmationPassword),
+              });
+            },
```

Comment:
Not needed right?
```javascript
onSubmit: (values) => accountController.updateUser({
                email: values.customer?.email || '',
                confirmationPassword: values.confirmationPassword,
              }),
```

Reply:
It is when `onSubmit` expects type `() => Promise<void>`. Since the `updateUser` returns a value, it complains. But this has to be refactored to consider the errors (wrapped with mutation or try/catch in the `onSubmit` handler)

Reply:
The try/catch is already in useForm. I think we only need to make more return types possible (type argument to useForm perhaps).

Reply:
This is not using `useForm` (yet) 😅

Reply:
Oops, that makes both my comments invalid 😅